### ISSUE TITLE
[Feature Req] Permission folders (website QoL)

### ISSUE THREAD

**Issue by JpegIsMe** [LuckPerms/LuckPerms#3732]
### Description

An optional folder feature to help organize permissions in the case where one group is filled with many permissions.
This would be exclusively for the website (/lp editor)

### Proposed Behaviour

Adding an optional grouping/folder method for permissions would help a lot with clutter in the case where one of your groups/ranks is filled to the brim with permissions.
This would not affect LP's main behaviour whatsoever, and would just be there as an optional feature to keep users organized in perm management.
(Adding an option to the `/lp [User/Group] set` command to add that permission in a specific folder would be a plus, but not necessary.)

### Extra Details

(Concept Image)
![image](https://github.com/LuckPerms/LuckPerms/assets/38381978/1e4e4ceb-e05c-4cf3-967e-2e5ab815ad2d)
Please consider this! It would make a lot of our lives easier!

**Comment by Turbotailz** [comment]
So... Groups

**Comment by JpegIsMe** [comment]
Not exactly, as the main purpose of this feature request would be to clean up clutter with large groups, instead of having to shovel through a giant list of sub-groups that you used to store/organize perms while giving players ranks
So instead of:
Groups:
  Builder
  Builder_Perms_Worldedit
  Builder_Perms_Misc

It would be:
Groups:
  Builder
    ∟ Worldedit
    ∟ Server Access
    ∟ Miscellaneous

The main purpose of this suggestion is quality of life. While there is a method to somewhat organize your permissions (Extra Groups), it isn't the greatest (Unless there's a feature LP has that I was never aware of).
